pkg/services/store/object/summary.go

```diff
@@ -0,0 +1,29 @@
+package object
+
+// ObjectSummary is derived from a RawObject and should not depend on system state
+// The summary is used for a unified search and listings objects since the fully
+type ObjectSummary struct {
+	UID         string            `json:"uid,omitempty"`
+	Kind        string            `json:"kind,omitempty"`
+	Name        string            `json:"name,omitempty"`
+	Description string            `json:"description,omitempty"`
+	Labels      map[string]string `json:"labels,omitempty"` // "tags" are represented as keys with empty values
+	URL         string            `json:"URL,omitempty"`    // not great to save here, but maybe not terrible :shrug:
+	Error       *ObjectErrorInfo  `json:"error,omitempty"`
+
+	// Optional values -- schema will define the type
+	Fields map[string]interface{} `json:"fields,omitempty"` // Saved as JSON, returned in results, but values not sortable
+
+	// eg: panels within dashboard
+	Nested []ObjectSummary `json:"nested,omitempty"`
+
+	// Optional references to external things
+	References []*ExternalReference `json:"references,omitempty"`
```

Comment:
I am not 100% sure if having the access to just the references is enough, that is:

```json
{
  "kind": "dashboard",
  "uid": 123,
  "references": [
    {
      "kind": "datasource",
      "type": "mysql",
      "uid": 10
    },
    {
      "kind": "datasource",
      "type": "postgress",
      "uid": 15
    },
    {
      "kind": "datasource",
      "type": "mongo",
      "uid": 20
    }
  ]
}
```

I think this skips some information, that is the relationship metadata, that we might want to access/search by

<img width="714" alt="image" src="https://user-images.githubusercontent.com/23451458/192291770-cc34c0df-581c-435b-8794-2d27d3047f8a.png">

===> 

```
{
  "kind": "dashboard",
  "uid": 123,
  "references": [
    {
      "relationshipType": "panelVar",
      "ref": {
        "kind": "datasource",
        "type": "mysql",
        "uid": 10
      }
    },
    {
      "relationshipType": "queryVar",
      "ref": {
        "kind": "datasource",
        "type": "postgress",
        "uid": 15
      }
    },
    {
      "relationshipType": "annotation",
      "ref": {
        "kind": "datasource",
        "type": "mongo",
        "uid": 20
      }
    }
  ]
}
```


this is pretty similar to the GraphQL connection conventions, see https://www.apollographql.com/blog/graphql/explaining-graphql-connections/#:~:text=Data-,on%20the%20edges,-Our%C2%A0UserFriendsConnection, "Data on the Edges" section. 

food for thought- its not necessarily something we have to do in this PR

Reply:
interesting -- yes we will want to identify what the reference is used for in addition to the reference.

Reply:
I suspect that it's not even sufficient to identify the type of the reference - each instance of a reference between each type of object needs a unique identifier. (That's the purpose of having the name `UserFriendsEdge` in the linked GraphQL docs).

But agreed that this needn't be resolved here - there are larger questions to be dealt with wrt references.